src/hooks/useClaims.ts

```diff
@@ -0,0 +1,67 @@
+import { useQuery } from "react-query";
+import { useConnection } from "./useConnection";
+import { ethers } from "ethers";
+import {
+  SUPPORTED_NETWORK_IDS,
+  contracts,
+  currentWindowIndex,
+  URLS,
+} from "../config";
+
+export type Claim = {
+  amount: string;
+  accountIndex: number;
+  metaData: string[];
+  windowIndex: number;
+  proof: string[];
+  ipfsHash: string;
+  rewardToken: string;
+  totalRewardDistributed: string;
+  hasClaimed: boolean;
+};
+export function useClaims() {
+  const { address, network } = useConnection();
+  const { data, isLoading, error, refetch } = useQuery<Claim[], Error>(
+    [address, network?.chainId],
+    () =>
+      getClaims({
+        address: ethers.utils.getAddress(address || ""),
+        chainId: network?.chainId as ChainId,
+      }),
+    {
+      enabled: Boolean(network?.chainId) && Boolean(address),
+    }
+  );
+
+  const filteredClaims = data?.filter(
+    (claim) => claim.windowIndex === currentWindowIndex
+  );
+
+  return { claims: filteredClaims, isLoading, error, refetch };
+}
+
+type ChainId = typeof SUPPORTED_NETWORK_IDS[number];
```

Comment:
Thoughts on exporting this from config or something, so it can be used everywhere instead of `1 | 42`?